anchor/message_validator/src/lib.rs

```diff
@@ -169,33 +174,159 @@ pub trait ValidatorService {
 }
 
 impl Validator {
-    pub fn new(processor: Senders, result_tx: Sender<Outcome>) -> Self {
+    pub fn new(
+        processor: Senders,
+        result_tx: Sender<Outcome>,
+        network_state_service: Arc<dyn NetworkStateService>,
+    ) -> Self {
         Self {
             processor,
             result_tx,
+            network_state_service,
         }
     }
 
-    fn do_validate(&self, _message: &SignedSSVMessage) -> Result<(), ValidationFailure> {
-        Ok(())
-    }
-
     fn validate_ssv_message(
         &self,
+        signed_ssv_message: &SignedSSVMessage,
         ssv_message: &SSVMessage,
     ) -> Result<ValidatedSSVMessage, ValidationFailure> {
         match ssv_message.msg_type() {
-            MsgType::SSVConsensusMsgType => QbftMessage::from_ssz_bytes(ssv_message.data())
-                .ok()
-                .map(ValidatedSSVMessage::QbftMessage)
-                .ok_or(ValidationFailure::UndecodableMessageData),
-            MsgType::SSVPartialSignatureMsgType => {
-                PartialSignatureMessages::from_ssz_bytes(ssv_message.data())
+            MsgType::SSVConsensusMsgType => {
+                let consensus_message = QbftMessage::from_ssz_bytes(ssv_message.data())
                     .ok()
-                    .map(ValidatedSSVMessage::PartialSignatureMessages)
-                    .ok_or(ValidationFailure::UndecodableMessageData)
+                    .ok_or(ValidationFailure::UndecodableMessageData)?;
+                self.validate_consensus_message_semantics(signed_ssv_message, &consensus_message)?;
+                Ok(ValidatedSSVMessage::QbftMessage(consensus_message))
             }
+            MsgType::SSVPartialSignatureMsgType => {
+                self.validate_partial_signature_message(ssv_message)
+            }
+        }
+    }
+
+    fn validate_partial_signature_message(
+        &self,
+        ssv_message: &SSVMessage,
+    ) -> Result<ValidatedSSVMessage, ValidationFailure> {
+        let messages = match PartialSignatureMessages::from_ssz_bytes(ssv_message.data()) {
+            Ok(msgs) => msgs,
+            Err(_) => return Err(ValidationFailure::UndecodableMessageData),
+        };
+
+        Ok(ValidatedSSVMessage::PartialSignatureMessages(messages))
+    }
+
+    fn validate_consensus_message_semantics(
+        &self,
+        signed_ssv_message: &SignedSSVMessage,
+        consensus_message: &QbftMessage,
+    ) -> Result<(), ValidationFailure> {
+        let signers = signed_ssv_message.operator_ids().len();
+
+        let committee_id = match signed_ssv_message.ssv_message().msg_id().duty_executor() {
+            Some(DutyExecutor::Committee(id)) => id,
+            _ => return Err(ValidationFailure::NonExistentCommitteeID),
+        };
```

Comment:
This is incomplete, I'm not handling the validator case. But this PR is already quite big, I will handle it in my next one which is coming soon.